scripts/linters/pylint_extensions.py

```
@@ -3021,36 +3011,145 @@ class PreventStringConcatenationChecker(checkers.BaseChecker):  # type: ignore[m
+            tokens: List[TokenInfo]. Object to access all tokens of a module.
+        """
+        # Formatted string literals (f-strings and t-strings) are tokenized
+        # as *_START and *_END tokens with the literal's content -- including
+        # any tokens inside replacement fields -- arriving in between. Track
+        # the literals in a stack (formatted strings can be nested) and only
+        # check the quote style once the closing token is reached, at which
+        # point it is known whether the literal contains a single quote.
+        formatted_string_stack: List[List[tokenize.TokenInfo]] = []
+        for token in tokens:
+            if token.type in (tokenize.FSTRING_END, tokenize.TSTRING_END):
+                if formatted_string_stack:
```

Ruff (0.15.21)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)